### ISSUE TITLE
1.3.0 Release

### ISSUE THREAD

**Issue by neersighted** [python-poetry/poetry#6972]
This issue tracks the progress of version 1.3.0 of Poetry. Please surface any blockers or concerns you have regarding this minor release here:

## Blockers

* Changelog
* Release notes (blog)
* poetry-plugin-export 1.2.0 (python-poetry/poetry-plugin-export#158)
* https://github.com/python-poetry/poetry/pull/6950

## Nice to have

* https://github.com/python-poetry/poetry/pull/6810
* ~~https://github.com/python-poetry/poetry-core/pull/517~~
* #6966
* #6824
* #6808 
* #6790 
* #6723 
* ~~#6709~~ 
* https://github.com/python-poetry/poetry-core/pull/512
* https://github.com/python-poetry/poetry-core/pull/487
* #6747
* ~~https://github.com/python-poetry/cleo/issues/248~~

## Important changes to surface

(This is intended to be the basis for the blog post and should cover new features or any manual steps/incompatibility when upgrading)

* The next release will turn off generation of `setup.py` by default (https://github.com/python-poetry/poetry-core/pull/318)

**Comment by finswimmer** [comment]
Nice to have: https://github.com/python-poetry/poetry/pull/6810

**Comment by dimbleby** [comment]
At time of writing, there are quite a lot of straightforward pull requests queued up.  I don't think any of them should block a release - I'm in favour of releasing more often! - but also I don't know any reason they shouldn't be merged and included.

Trying to err on the side of caution, I'd still suggest that the following look ready to go:

- #6966
- #6824
- #6808 
- #6790 
- #6723 
- #6709

**Comment by dimbleby** [comment]
#6747 is maybe also desirable (needs a poetry-core release first)

**Comment by neersighted** [comment]
Edited; I'll start taking a look at finishing/merging the low hanging fruit later today.

**Comment by rickzinit** [comment]
any possibility to release v.1.3.0 this week?

**Comment by neersighted** [comment]
As ever, soon/when we can.

**Comment by neersighted** [comment]
Regarding #6709, without CI to validate it, and with other work needed to migrate Poetry to the (eminent) 1.0 release, I am inclined to defer it to reduce risk and my need to manually validate the change.

https://github.com/python-poetry/cleo/issues/248 would be nice, but we can bump Cleo in a patch release and @Secrus has limited bandwidth to get #6709 reviewed/Poetry rebased on top of the breaking changes in Cleo 1.0.

**Comment by neersighted** [comment]
The status of 1.3 looks like this: we need a core release, then we need to rebase one PR against Poetry on our "nice-to-have" list to account for that. We can then review & merge that PR, prepare a version bump, and draft our release notes/CHANGELOG.

**Comment by dimbleby** [comment]
re cleo, I have code sitting around locally that fixes up the remaining type errors if and when the cleo release happens.

So if cleo _were_ to release 1.0 soon, and if y'all did want to get that into 1.3.0 - then it should be possible to move fast.

I don't know that that's especially desirable - I can think of a couple of useful bug-fixes, but nothing that couldn't wait for a 1.3.1 or whatever - and certainly I wouldn't advocate blocking poetry 1.3.0 on cleo.

**Comment by neersighted** [comment]
That's pretty much where it's at -- there's not benefit to the partial approach before 1.0 is ready, and Cleo 1.0 is mostly blocked on bandwidth. What we have works, let's ship that and then we can use automated tooling to make changes for Cleo 1.0 with more confidence once it is tagged; then backport it if desired to make life easier to the distro packagers.

**Comment by dimbleby** [comment]
Just so we're clear, I'm saying that Secrus bandwidth to rebase poetry on cleo 1.0 breaking changes need not be a factor, I have that done already.

(again, I'm neverthless fully ok with not waiting for cleo)

**Comment by finswimmer** [comment]
If we have the time I would like to see #7100 in 1.3 as there are already other fixes regarding `virtualenvs.prefer-active-python`.

**Comment by dimbleby** [comment]
Ship it!

I'm not sure what it is that causes poetry's release-paralysis - is the release process burdensome?  is there anything the rest of us can do to make it not burdensome?

So far as I can see the codebase has been in a sensible state to cut a release for quite some time; let's do it, and move on to the next one.

(There's only so many times I can tell people to clear their cache, releasing the fixes is a much more scalable solution...)
